### PR TITLE
Handle app seating in app manager properly

### PR DESCRIPTION
**Pull request by amstone326**
Was doing a couple of things carelessly in regards to app seating in the app manager. This PR makes it so that:
-For operations in `SingleAppManagerActivity` that require the current app to be seated (updating and verifying MM), we only try to seat the app if it is not already seated.
-When seating the current app is necessary, use the `SeatAppActivity` rather than just calling 'initializeAppResources()` (for the same reason we decided to use the SeatAppActivity on the login screen -- the seating process can take a while for some apps, and having the screen just go unresponsive with no indication of what's happening is a terrible UX).

### CODE REVIEW

File: app/src/org/commcare/activities/SingleAppManagerActivity.java [hunk 1]
```diff
@@ -116,7 +122,7 @@ private void refresh() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
         switch (requestCode) {
-            case CommCareHomeActivity.UPGRADE_APP:
```
Comment (phillipm) [inline app/src/org/commcare/activities/SingleAppManagerActivity.java]: Can you remove the definition of `CommCareHomeActivity.UPGRADE_APP`?

File: app/src/org/commcare/activities/SingleAppManagerActivity.java [hunk 2]
```diff
@@ -231,7 +254,21 @@ public void updateClicked(View v) {
...
+    private void seatApp() {
+        Intent i = new Intent(this, SeatAppActivity.class);
+        i.putExtra(SeatAppActivity.KEY_APP_TO_SEAT, appRecord.getUniqueId());
+        this.startActivityForResult(i, SEAT_APP_ACTIVITY);
+    }
+
+    private void launchUpdateActivity() {
         Intent i = new Intent(getApplicationContext(), UpdateActivity.class);
         i.putExtra(AppManagerActivity.KEY_LAUNCH_FROM_MANAGER, true);
         startActivityForResult(i, CommCareHomeActivity.UPGRADE_APP);
```
Comment (phillipm) [inline app/src/org/commcare/activities/SingleAppManagerActivity.java]: Shouldn't this change to `UPGRADE_APP`?

File: app/src/org/commcare/activities/SingleAppManagerActivity.java [hunk 3]
```diff
@@ -203,7 +218,15 @@ public void verifyResourcesClicked(View v) {
...
+            seatApp();
+        } else {
+            launchVerificationActivity();
+        }
+    }
+
+    private void launchVerificationActivity() {
         Intent i = new Intent(this, CommCareVerificationActivity.class);
         i.putExtra(AppManagerActivity.KEY_LAUNCH_FROM_MANAGER, true);
         this.startActivityForResult(i, DispatchActivity.MISSING_MEDIA_ACTIVITY);
```
Comment (phillipm) [inline app/src/org/commcare/activities/SingleAppManagerActivity.java]: Shouldn't this reference `SingleAppManagerActivity.MISSING_MEDIA_ACTIVITY`?

File: app/src/org/commcare/activities/SingleAppManagerActivity.java [hunk 4]
```diff
@@ -35,6 +35,12 @@
     private static final int LOGOUT_FOR_VERIFY_MM = 1;
     private static final int LOGOUT_FOR_ARCHIVE = 2;
 
+    private static final int UPGRADE_APP = 0;
+    private static final int MISSING_MEDIA_ACTIVITY = 1;
+    private static final int SEAT_APP_ACTIVITY = 2;
+
+    boolean launchUpdateAfterSeating;
```
Comment (phillipm) [inline app/src/org/commcare/activities/SingleAppManagerActivity.java]: `private`

File: app/src/org/commcare/activities/SingleAppManagerActivity.java [hunk 5]
```diff
@@ -231,7 +254,21 @@ public void updateClicked(View v) {
      * Conducts an update for the selected app
      */
     private void update() {
-        CommCareApplication._().initializeAppResources(new CommCareApp(appRecord));
+        if (!CommCareApplication._().isSeated(appRecord)) {
+            launchUpdateAfterSeating = true;
```
Comment (phillipm) [inline app/src/org/commcare/activities/SingleAppManagerActivity.java]: What happens if you rotate the device while seating the app?

**Comment by amstone326** [comment]
Yikes this was a little sloppy, thanks for the checks. Fixed now.

**Pull request merged**